packages/gatsby/src/schema/print.js

```diff
@@ -218,12 +218,16 @@ const printObjectType = tc => {
 
 const printInterfaceType = tc => {
   const type = tc.getType()
+  const interfaces = type.getInterfaces()
+  const implementedInterfaces = interfaces.length
+    ? ` implements ` + interfaces.map(i => i.name).join(` & `)
+    : ``
```

Comment:
NIT: given we have https://github.com/gatsbyjs/gatsby/blob/3e82e72d016614ce5b038bd455c82f6af7e70c7d/packages/gatsby/src/schema/print.js#L198-L200 already used for object types - maybe just have this moved to shared `printImplementedInterfaces` helper function?

(not huge deal, up to you ;) )

Reply:
I usually follow [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) for cases like this (that's why I didn't extract it to a separate utility)

Reply:
One example of how this can be useful in this specific case - imagine they'll change `interface Foo implements Bar` to something like `interface Foo extends Bar` but will keep `type Foo implements Bar`. In this case, the current version is much easier to refactor.

 This stuff happens all the time with shallow abstractions like this.

Reply:
Fair, let's keep it as-is :)